hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/JpaTargetManagement.java

```diff
@@ -851,6 +855,36 @@ public boolean existsByControllerId(final String controllerId) {
         return targetRepository.exists(TargetSpecifications.hasControllerId(controllerId));
     }
 
+    @Override
+    public boolean isDeviceMatchingTargetFilterQueryAndNonDSAndCompatible(final String controllerId,
+            final long distributionSetId, final String targetFilterQuery) {
+        if (!isTargetFilterQueryValid(targetFilterQuery)) {
+            return false;
+        }
+        return distributionSetManagement.get(distributionSetId).map(ds -> {
```

Comment:
I would rather throw `EntityNotFoundException` in case DS could not be found as it is not a normal operational flow and should be considered as error